plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSource.java

```diff
@@ -90,12 +122,22 @@ public Page getNextPage()
                 return null;
             }
 
+            BlockBuilder rowIndices = new LongArrayBlockBuilder(null, batchSize);
+            for (long position = 0; position < batchSize; position++) {
+                rowIndices.writeLong(parquetReader.lastBatchStartRow() + position);
+            }
+
             Block[] blocks = new Block[fields.size()];
-            for (int fieldId = 0; fieldId < blocks.length; fieldId++) {
-                Type type = types.get(fieldId);
-                blocks[fieldId] = fields.get(fieldId)
-                        .<Block>map(field -> new LazyBlock(batchSize, new ParquetBlockLoader(field)))
-                        .orElseGet(() -> RunLengthEncodedBlock.create(type, null, batchSize));
+            for (int column = 0; column < blocks.length; column++) {
+                if (isIndexColumn(column)) {
+                    blocks[column] = rowIndices.build();
```

Comment:
It's OK to build `rowIndices` as many times as many row index columns there are. We don't need to optimize for a theoretical case when this column is requested twice or more.